AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -42,4 +49,60 @@ - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
   return [ASLayout layoutWithLayoutableObject:self size:constrainedSize.min];
 }
 
+- (ASLayoutSpec *)finalLayoutableWithParent:(ASLayoutSpec *)parentSpec;
+{
+  return nil;
+}
+
+- (void)setChild:(id<ASLayoutable>)child;
+{
+  [self setChild:child forIdentifier:kDefaultChildKey];
+}
+
+- (id<ASLayoutable>)layoutableToAddFromLayoutable:(id<ASLayoutable>)child
+{
+  ASLayoutOptions *layoutOptions = [child layoutOptions];
+  
+  id<ASLayoutable> finalLayoutable = [child finalLayoutableWithParent:self];
+  if (finalLayoutable) {
+    [layoutOptions copyIntoOptions:finalLayoutable.layoutOptions];
```

Comment:
Create layoutOptions local variable only inside this conditional, as it appears there is a case where it is not needed (e.g. finalLayoutable is nil).  I don't have an intuitive understanding of this code, though, and think it deserves a comment above or at the start of the method.  In what scenario is it called, and what are its semantics?  Is this the thing that currently handles propagating up properties such as baseline alignment for inset-wrapped stack children?